### PR TITLE
feat: add support to parse hive genesis file.

### PR DESCRIPTION
**Pull request by mpaulucci**
**Motivation**
We want to be able to run hive simulations

**Description**
Some fields are slighly different in hive compared to kurtosis. For example:

-`timestamp`, `nonce` is a dec string whereas it is a hex string in the kurtosis file
- `mix_hash` comes as `mixhash` in kurtosis but `mixHash` in hive

Closes part of #205

### CODE REVIEW

File: crates/core/serde_utils.rs [hunk 1]
```diff
@@ -10,7 +10,8 @@ pub mod u256 {
         D: Deserializer<'de>,
     {
         let value = Number::deserialize(d)?.to_string();
-        U256::from_dec_str(&value).map_err(|e| D::Error::custom(e.to_string()))
+        U256::from_dec_str(&value)
+            .map_err(|e: ethereum_types::FromDecStrErr| D::Error::custom(e.to_string()))
```
Comment (fmoletta) [inline crates/core/serde_utils.rs:14]: is this needed?

Comment (mpaulucci) [reply crates/core/serde_utils.rs:14]: no idea how that got there

File: cmd/ef_tests/types.rs [hunk 2]
```diff
@@ -177,12 +177,11 @@ pub struct Transaction {
     pub blob_versioned_hashes: Option<Vec<H256>>,
     pub hash: Option<H256>,
     pub sender: Address,
-    #[serde(deserialize_with = "crate::serde_utils::h160::deser_hex_str")]
+    #[serde(deserialize_with = "ethereum_rust_core::serde_utils::h160::deser_hex_str")]
     pub to: Address,
```
Comment (fmoletta) [inline cmd/ef_tests/types.rs:181]: We could change the type of `to`  to `TxKind` as it already has `Deserialize` implemented

File: crates/core/serde_utils.rs [hunk 3]
```diff
@@ -30,6 +30,48 @@ pub mod u256 {
...
+        D: Deserializer<'de>,
+    {
+        let value = String::deserialize(d)?;
+        if value.is_empty() {
+            Ok(H160::zero())
+        } else {
+            H160::from_str(value.trim_start_matches("0x"))
+                .map_err(|_| D::Error::custom("Failed to deserialize H160 value"))
+        }
+    }
```
Comment (fmoletta) [inline crates/core/serde_utils.rs:74]: This implementation is misleading, while this is the behaviour we want for a `TxKind` field where an empty string is a valid value, we wouldn't want to use it to deserialize an address or H160 value

File: crates/core/serde_utils.rs [hunk 4]
```diff
@@ -30,6 +30,48 @@ pub mod u256 {
...
         U256::from_dec_str(&value).map_err(|e| D::Error::custom(e.to_string()))
     }
+
+    pub fn deser_hex_str<'de, D>(d: D) -> Result<U256, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = String::deserialize(d)?;
+        U256::from_str_radix(value.trim_start_matches("0x"), 16)
+            .map_err(|_| D::Error::custom("Failed to deserialize u64 to u256 value"))
```
Comment (fmoletta) [inline crates/core/serde_utils.rs:40]: This error doesn't sound right

File: crates/core/serde_utils.rs [hunk 5]
```diff
@@ -30,6 +30,48 @@ pub mod u256 {
...
+    }
+
+    pub fn deser_hex_or_dec_str<'de, D>(d: D) -> Result<U256, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let value = String::deserialize(d)?;
+        if value.starts_with("0x") {
+            U256::from_str_radix(value.trim_start_matches("0x"), 16)
+                .map_err(|_| D::Error::custom("Failed to deserialize u64 to u256 value"))
```
Comment (fmoletta) [inline crates/core/serde_utils.rs:50]: This error doesn't sound right

**Review by fmoletta — APPROVED** [review]
LGTM

**Pull request merged**